src/app/api/auth/auth.ts

```diff
@@ -0,0 +1,54 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { decrypt } from '@/utils/encryption';
+import { jwtDecode } from 'jwt-decode';
+import { Account, getServerSession } from 'next-auth';
+import type { JWT } from 'next-auth/jwt';
+import { ExtendedSession } from './auth.types';
+import { authOptions } from './config';
+
+export async function getToken(tokenType: 'access_token' | 'id_token') {
+  const session = (await getServerSession(authOptions)) as ExtendedSession;
+  if (session) {
+    const tokenDecrypted = decrypt(session[tokenType]!);
+    return tokenDecrypted;
+  }
+  return null;
+}
+
+export function completeTokenWithAccountInfo(token: JWT, account: Account): JWT {
+  return {
+    ...token,
+    decoded: jwtDecode(account.access_token!) as string,
+    access_token: account.access_token as string,
+    id_token: account.id_token as string,
+    refresh_token: account.refresh_token as string,
+    expires_at: account.expires_at as number,
+  };
+}
+
+export async function refreshAccessToken(token: JWT) {
```

Comment:
wasn't this the same function that was in src/utils/auth.ts that was deleted? what was the problem then for the session to expire?

Reply:
I also don't understand exact fix here, seems like more or less same logic moved around. Is it related to next?

Reply:
Sorry I did some refactoring as well so the important changes have been drowned in all these changes.

I made a comment to show you the fix